### PR TITLE
Blog article

### PR DESCRIPTION
**Pull request by Lukewh**
# Done

Fixes https://github.com/canonical-websites/snapcraft.io/issues/702

- Simple blog article page
- Author to blog list items
- SEO metadata
- Related posts
- Tests

# QA

- Pull the branch
- `./run`
- Visit http://0.0.0.0:8004/blog
- Visit a few articles and see that the formatting is OK
- Made naming consistent in the code - `article` & `articles`

# Screenshot

screenshot_2018-06-15 decreasing the complexity of iot adoption with edge as a service model snapcraft 1

### CODE REVIEW

**Comment by bartaz** [comment]
Layout of some blog posts seems weird (with headings wider then text content):
screen shot 2018-06-14 at 16 52 55
http://snapcraft.io-pr-734.run.demo.haus/blog/circleci-evangelist-looks-to-snaps-to-bridge-cross-distro-divide

Maybe it should be limited to a grid (as on blog.ubuntu.com)

**Comment by therealjuan** [comment]
Might be good to align the `<code>` style with the one we have on the install details section under snap details page.

Also, might be good to update the table styling for the blog when there is an image and include a right padding to avoid this:

image

**Comment by bartaz** [comment]
Probably not an issue with this particular PR, but /blog takes quite a long time to load - 5sec for me. Is the API that slow?

screen shot 2018-06-15 at 11 22 34

**Comment by Lukewh** [comment]
@bartaz it's because for each post we have to make separate requests to get the featured image and the author details (for the name)... Because Wordpress believe in pure REST, rather than useable REST.

Before going live we definitely need to make sure it's quicker through caching or some other way.

**Comment by Lukewh** [comment]
@therealjuan the problem with doing `<code>` is that they can be really long multi-line. We should figure out a better design/ see what other sites do?

File: webapp/blog/logic.py [hunk 1]
```diff
@@ -37,8 +39,17 @@ def transform_article(article, featured_image=None):
...
+        # split at the last 3 characters
+        raw_article_start = raw_article[:-3]
+        raw_article_end = raw_article[-3:]
+        # for the last 3 characters replace any part of […]
+        raw_article_end = raw_article_end.replace('[', '')
+        raw_article_end = raw_article_end.replace('…', '')
+        raw_article_end = raw_article_end.replace(']', '')
+        # join it back up
+        raw_article = raw_article_start + raw_article_end
+        article['excerpt']['raw'] = raw_article + ' […]'
```
Comment (tbille) [inline webapp/blog/logic.py]:
```python
''.join([
  raw_article,
  ' […]',
])
```

**Pull request merged**